Dockerfile

```
@@ -1,17 +1,17 @@
-FROM postgis/postgis:15-3.5 AS development_build
+FROM ghcr.io/baosystems/postgis:15-3.5 AS development_build
```

@amandasaurus had already flagged this change and I was wondering about the reasoning for this one too! 

I've run it with this on my (ubuntu 24) machine, and it seems to work fine with the original image too: 

```suggestion
FROM postgis/postgis:15-3.5 AS development_build
```

My understanding is that the changed image is [this one](https://github.com/baosystems/docker-postgis), which adds support for mixing 64 & 32 bit architectures? I'm not sure if that's needed for this repo? My main reason for sticking to the official one would be this in the fork: 

> No support is provided. If you do not understand the implications of using this image, do not use it and use the [official PostGIS image](https://hub.docker.com/r/postgis/postgis) instead.

I'm quite a newcomer to docker myself (I only have what in German we call "dangerous half-knowledge" :joy:), so if I miss a reason for the switch, definitely happy to learn more! :slightly_smiling_face:

---

Yes, this is a good thing to flag! I should have spent more than a few words in the commit message talking about it.

I switched to this other image because the default `postgis/postgis` image is not multi-architecture compatible. It's only built for `amd64` and not also `arm64`. This means I would have to use emulation on my MacBook M1 to run it. This is possible, but much more cumbersome.

There is a years-old issue to resolve this in the repository that builds `postgis/postgis` (https://github.com/postgis/docker-postgis/issues/216), but it hasn't closed yet. In the meantime, someone set up a fork of that repository and switched to using GitHub Workflows to build the image and publish it to GitHub's image service (the `ghcr.io` part in the new image name). As part of that change, they also enabled multi-arch support and built an arm64 image. Everything else is the same in that image.

If you're still not interested in this change, I'm happy to switch the image back and keep this change in my fork only.

---

> This is possible, but much more cumbersome.

@iandees What issues do you experience when running the emulated `postgis/postgis` image? I didn't have any issues on my Apple silicon Mac, neither did I have to install Rosetta in order to run it.

It's a pity that the official image is not built for arm64, but I still don't see a reason for not sticking to it.

---

Ah, that's embarrassing. I've actually run into this issue with M1 chips in the past myself :joy: 

Back then we solved it through emulation by adding 

```
# If this is an M1 mac, make sure we use docker QEMU amd64
if [[ $(uname -m) == 'arm64' ]]; then
  echo "Running on an M1 mac; setting to use amd64 images under QEMU"
  export DOCKER_DEFAULT_PLATFORM=linux/amd64
fi
```

to the `docker_run.sh` to enable the emulation. Is that what you were referring to as well? 

I don't have access to an M1 Mac to test how bad the performance is using the emulation, so I don't have strong feelings for either approach, but reading through the GH issue you linked @iandees, it looks to me like a solid solution until there's an official postgis image to use. 

Just to double check: Did you try if the image also works in MyBinder? :)

---

I was trying to avoid the warnings

```
 ---> [Warning] The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
```

... but you're right, those are only warnings and the existing image runs fine. It feels a bit slower, but still adequate.

---

:+1: In that case, I think it would be nicer to keep the "_official_" image. I do not doubt that the fork to make it actually arm compatible is trustworthy, but if we start offering people a way to log into their OSM accounts (c.f. #8), then I'd feel a bit more at ease if we didn't rely on a forked image as the base. Does that make sense? :slightly_smiling_face:

---

Yep! Resolved in f7f365e.

---

Perfect, thanks so much!